website/docs/sql/scalar_func/arrays.md

```diff
@@ -0,0 +1,53 @@
+# Array Functions
+
+This section describes functions and operators to inspect and transform `array` values. In the following, `array(T)` denotes an array type with element type `T`. 
+The element type is omitted when irrelevant.
+
+
+## Access & Inspection
+The following functions yield access to array elements or metadata.
+These operations are compatible with all (nullable and non-nullable) element types.
+
+:::note
+Some of the functions listed below may seem nonsensical, as they only return static values (e.g., `array_lower`). These functions exist mainly for compatibility with Postgres, which allows multi-dimensional arrays with non-uniform indices.
+:::
+
+Signature|Description|Example
+---|---|---
+<code>array(T)**[**int**]**</code> → `T`| Returns the n-th element of the array (1-indexed). | `(array[1,2,3])[1]` → `1`
+<code>array(T)**[**int**:**int**]**</code> → `T` | Returns the subarray within the given boundes (1-indexed, inclusive). |`(array[1,2,3])[2:3]` → `{2,3}` | 
+<code>**array_length(**array**)**</code> → `int` | Returns the length of the array. | `array_length(array[1,2,3])` → `3`
+<code>**array_dims(**array**)**</code> → `text` | Returns a textual description of the start and end index for each dimension of the array (always <code>[1:_length_]</code>).| `array_dims(array[1,2,3])` → `[1:3]`
+<code>**array_ndims(**array**)**</code> → `int`| Returns the numbers of dimensions in the array (always `1`).| `array_ndims(array[1,2,3])` → `1`
+<code>**array_lower(**array**)**</code> → `int` | Returns the index of the first element in the array (always `1`). |`array_lower(array[1,2,3])` → `1`
+<code>**array_upper(**array**)**</code> → `int` | Returns the index of the last element in the array (always equal to `array_length(array)`). |`array_upper(array[1,2,3])` → `3`
+<code>**array_to_string(**array, text [, text]**)**</code>| Converts the array into a textual representation, with the given element separator and (optional) null indicator. | `array_to_string(array[1,2,3], ';')` → `1;2;3`<br/>`array_to_string(array[3,2,1,null], '⏰', '🎉')` → `3⏰2⏰1⏰🎉` 
+
+## Transformations
+The following functions produce new array values from existing ones.
+These operations are compatible with all (nullable and non-nullable) element types.
+
+Signature|Description|Example
+---|---|---
+<code>**array_prepend(**T,array(T)**)**</code> → `array(T)`<br/>or<br/><code>T **\|\|** array(T)</code> → `array(T)`| Inserts a value at the front of the array.| `array_prepend(1, array[2,3,4])` → `array[1,2,3,4]` 
+<code>**array_append(**array(T), T**)**</code> → `array(T)`<br/>or<br/><code>array(T) **\|\|** T</code> → `array(T)`| Inserts a value at the end of the array.| `array_append(array[1,2,3], 4)` → `array[1,2,3,4]` 
+<code>**array_concat(**array(T), array(T)**)**</code> → `array(T)`<br/>or<br/><code>array(T) **\|\|** array(T)</code> → `array(T)`| Concatenates two arrays. | `array_concat(array[1,2], array[3,4])` → `array[1,2,3,4]`  
```

Comment:
Concat can also concatenate elements and arrays.

Reply:
We were both wrong: the SQL function is called `array_cat`, but it only operates on array operands.
[9fabdcd](https://github.com/tableau/hyper-db/pull/110/commits/9fabdcd8f563d54f052cc80085c08c87b54f630a)

Reply:
But `||` can also concatenate elements and arrays

Reply:
Yes, but that's already documented ;)

![Screenshot 2024-08-06 at 10 50 30](https://github.com/user-attachments/assets/fe86a3cb-4e4b-4803-badc-6000c92b3825)